### ISSUE TITLE
Error in initialize_tree of SCUBA

### ISSUE THREAD

**Issue by ldorozco** [gcyuan/SCUBA#5]
Hi, 

I get an error when running SCUBA. I get the same error independent of what method I use ('pca', 'pac2', 'clustermode') Do yo know what could be causing this? I am using fluidigm data. 

I only get this error when I use my own data set, not the guo2010 sample data set.  However the format of my data is the the same as the guo2010 sample data. Below is the head of my file. 

ERROR: 
Error in initialize_tree (line 87)
y.clu_idx(I) = idx;

Error in SCUBA (line 36)
initialize_tree(dataset, cluster_mode);

File head:

Cell ID 1   2   3   4   5
ITGAE   31.00   31.00   31.00   31.00   31.00
CD3E    12.39   11.35   11.61   11.87   11.98
CD101   31.00   31.00   14.62   31.00   14.95
GZMB    31.00   17.72   17.81   16.78   11.92
TGFB1   15.19   31.00   31.00   17.11   14.00
KLRC2   21.13   31.00   17.94   22.02   19.74
GZMA    31.00   31.00   31.00   31.00   13.96
CD4 13.67   15.85   14.00   31.00   31.00
CD160   31.00   31.00   31.00   31.00   31.00

Thanks so much in advance. 

Luz

**Comment by gcyuan** [comment]
Hi Luz,

Thanks for using scuba. I cannot say for sure what the problem might be since I don't have the data. Just a hunch, is this a qPCR dataset where the maximum value is 31.00? If so, you should convert it to gene expression level first by subtracting 31 and remove those genes that are not expressed in any cells.

Let me know if that helps.

GC

**Comment by eugeniomarco** [comment]
Hi Luz,

I would add that for some reason it seems that idx is not defined or empty.
idx is calculated in line 81 in initialize_tree.m. It might be easier to
trace the error by looking first at what happens in that line. If you are
familiar with matlab you can place a breakpoint there and see the inputs
and outputs to kmeans that cause the error. Without having the data it's a
bit difficult to trace the error.

Hope that helps.

Regards
Eugenio

**Comment by GGiecold-zz** [comment]
Dear Luz,

I believe I have identified the source of your issue.

Based on the 'file head' you provided along with your error log, it
seems that your dataset does not feature any temporal information.
Assuming you are dealing with qPCR samples, please run the following
Matlab command for pseudotime inference:

> > PCR_preprocess('my_super-duper_data');

I would advise running this command anyway: even if your dataset
consists in proper time series recordings, 'PCR_preprocess',
'RNAseq_preprocess' and 'MassCytometry_preprocess' perform some
transformations such as mean-variance normalization (check the
documentation for details on what kind of transformations are actually
performed; this depends on your data-type and on some default parameters).

After this preliminary processing, proceed as before by running the
SCUBA procedure with your favorite choice of cluster_mode parameter
(i.e. 'pca', 'pca2' or 'euclidean'):

> > SCUBA('my_super-duper_data', cluster_mode)

Please let me know if this helps solving your problem. Note that I wrote
a Python implementation of the SCUBA algorithm. It's been dubbed PySCUBA
and is faster and more reliable (found some mistakes in the Matlab
code). It should be released quite soon.

Best regards,

Gregory

**Comment by ldorozco** [comment]
Hi,

Thanks to all for your responses. The problem was indeed in my input file, in that it did not contain a row for 'Stage'. Running "PCR_preprocess" function made no difference but once I added the line to my input file it was able to get pass that problem. I also converted the Ct values to expression values but I don't think this makes a difference since the sample_data "guo2010Data.txt"  seem to also have unconverted Ct values. 

Unfortunately I then ran into another issue which I will post. 

Thanks so much for your help. 

-Luz
